Put driver files block

vkz-mlx opened the pull request:
Adding the RC.1.3.0 as a base version for MLX75322 SPI driver

vkz-mlx committed:
Put driver files block

kravchenkoloznia commented on Make-75322-raspi.mk:
```diff
@@ -0,0 +1,26 @@
+# Makefile for customizing build for 75322 (Raspberry Pi-pased compilation)
```
typo `Pi-pased`

kravchenkoloznia replied:
Based on description that two files: `Make-75322-raspi.mk` and `Make-75322.mk`; used to customize build for the same product. So why we need both of them?

vkz-mlx replied:
customization and further family growth. Each file - each platform.

kravchenkoloznia commented on Make-75322.mk:
```diff
@@ -0,0 +1,25 @@
+# Makefile for customizing build for 75322 (PC-pased compilation)
```
typo `PC-pased`

vkz-mlx replied:
Fixed

kravchenkoloznia commented on Make-75322.mk:
```diff
@@ -0,0 +1,25 @@
+RELEASE_MATURITY = RC
+RELEASE_MAJOR    = 1
+RELEASE_MINOR    = 3
+RELEASE_REVISION = 0
```
It seems like it's duplicating release version from `Make-75322-raspi.mk`

vkz-mlx replied:
Moved to root Makefile

kravchenkoloznia commented on Makefile:
```diff
@@ -0,0 +1,147 @@
+CPPFLAGS += -fpic
+CPPFLAGS += -std=c99
+CPPFLAGS += -c -g -ffunction-sections -fdata-sections -fms-extensions\
+	-O -Wall -W
```
what is the reason to add `\` symbol instead of add another line with `CPPFLAGS +=`

vkz-mlx replied:
One-line assignment used

kravchenkoloznia commented on doc/Makefile:
```diff
@@ -0,0 +1,465 @@
+	@$(ECHO) "  html       to make standalone HTML files"
+	@$(ECHO) "  singlehtml to make a single large HTML file"
+	@$(ECHO) "  latex      to make LaTeX files, you can set PAPER=a4 or PAPER=letter"
+	@$(ECHO) "  latexpdf   to make LaTeX files and run them through pdflatex"
```
`doxy` target is not mentioned in help

kravchenkoloznia commented on doc/Makefile:
```diff
@@ -0,0 +1,465 @@
+DOXYGEN_INPUT += $(addsuffix /src, $(LIBRARY_DIRS))
+DOXYGEN_INPUT += $(addsuffix /inc, $(LIBRARY_DIRS))
+.PHONY: doxy
+doxy: $(BUILDDIR)/html/doxygen
+	@$(ECHO) "** Running doxygen on source code, destination of doxygen build is the final html folder."
+	$(HIDE_CMD) DOXYGEN_INPUT="$(DOXYGEN_INPUT)" COMPONENT_FLAGS="$(COMPONENT_CPPFLAGS)" doxygen doxygen/doxyfile $(NO_OUTPUT)
```
do we really need all other stuff in that **Makefile** if only `doxy` target is used?

vkz-mlx replied:
Significantly cleaned-up

kravchenkoloznia commented on doc/source/conf.py:
```diff
@@ -0,0 +1,448 @@
+# autogenerated file.
+#
+# All configuration values have a default; values that are commented out
+# serve to show the default.
```
That file maybe deleted if only doxygen documentation will be used.

kravchenkoloznia commented on pip-dependencies.txt:
```diff
@@ -0,0 +1,20 @@
+--extra-index-url https://pypi.org/simple
+
+# Pip packages needed in general
+mlx.warnings>=0.2.0
```
Is it used in that project?

vkz-mlx replied:
whole file removed as not required

kravchenkoloznia commented on pip-dependencies.txt:
```diff
@@ -0,0 +1,20 @@
+--extra-index-url https://dsl.melexis.com/artifactory/api/pypi/swcc-pypi-release/simple
+--extra-index-url https://dsl.melexis.com/artifactory/api/pypi/pypi-release/simple
+--extra-index-url https://pypi.org/simple
```
If `mlx.warnings` plugin is not used, extra index url's to `dsl.melexis.com` is not needed

vkz-mlx replied:
removed as described above https://github.com/melexis/mlx75322-spi-driver/pull/1#discussion_r382599563

vkz-mlx committed:
clean-up Makefile for documentation

vkz-mlx committed:
Fixing typo

vkz-mlx committed:
clean-up conf.py

dryodon commented on doc/doxygen/footer.html:
```diff
@@ -0,0 +1,24 @@
+    $navpath
+    <li class="footer">
+    <img class="footer" src="../images/melexis-logo.png" alt="Melexis N.V," height="25px" hspace="10"/>
+	&copy; Copyright 2015 Melexis N.V. ---
```
2015?

vkz-mlx replied:
Updated

kravchenkoloznia reviewed (commented):
Some small comments left.

dryodon commented:
Please explain in readme on what systems you built the driver and what you needed to do to build it.
Stress that you're not sure nor guaranteeing that it builds on other systems.

vkz-mlx committed:
Review's updates:
  - Put release version in root Makefile;
  - clean-up uncrustify(stylechecker) target;
  - put CPPFLAGS in one line, w/o '\'

vkz-mlx committed:
clean-up `RELEASE_*` variables

vkz-mlx committed:
Readme as initial info input updated

vkz-mlx committed:
Fix footer 'Copyright 2020' year

vkz-mlx committed:
remove "pip-dependencies.txt" as explicit

vkz-mlx committed:
update .gitignore to omit the `build/`folder changes tracking

vkz-mlx committed:
Make 75322-raspi as the default configuration. Update desc and Makefile
